tests/test_compute_utils.py

```diff
@@ -373,6 +374,46 @@ def test_surface_averages_shortcut(self):
             err_msg="average without sqrt(g) fail",
         )
 
+    @pytest.mark.unit
+    def test_surface_variance(self):
+        """Test that variance is computed correctly when weights are equal."""
+        grid = LinearGrid(L=L, M=M, N=N, NFP=NFP)
+
+        # actual number of samples per surface
+        n = grid.num_rho * grid.num_theta
+        np.testing.assert_allclose(np.bincount(grid.inverse_zeta_idx), desired=n)
+
+        # something arbitrary
+        q = np.arange(grid.num_nodes)
+        # Check correctness of mean calculation.
+        mean = grid.expand(
+            q.reshape((grid.num_zeta, -1)).mean(axis=-1),
+            surface_label="zeta",
+        )
```

Comment:
Our predefined grids sort the nodes in chunks of common zeta coordinates. So to compute a quantity that is local to a surface, we can reshape them into zeta surface chunks and take the mean across the chunks.